src/gtest/tx_creation_utils.cpp

```diff
@@ -668,15 +668,15 @@ MempoolReturnValue BlockchainTestManager::TestAcceptTxToMemoryPool(CValidationSt
 {
     CCoinsViewCache* saved_pcoinsTip = pcoinsTip;
 
-    std::unique_ptr<CTxMemPool> pool(new CTxMemPool(::minRelayTxFee, DEFAULT_MAX_MEMPOOL_SIZE_MB * 1000000));
```

Comment:
While I totally agree with modification (from global `mempool` to local `pool`) contained in `main.cpp`, I think this function should be left unchanged; better to use locally created `pool`, instead of resetting each time global `mempool`.

What I am proposing is supported by:

- other tests in this module using the same approach (`TxInputLimit`, `SproutV3TxFailsAsExpected`, `SproutV3TxWhenGrothNotActive`, `SproutNegativeVersionTx`)
- just tried running all gtests on my machine and confirm what is written above


Even if it can be worth to have a deeper understanding of the failure you encountered, with my proposal there is no need to modify tests like `src/wallet/gtest/test_wallet_zkeys.cpp` and `src/wallet/gtest/test_deadlock.cpp`.

Reply:
Done

Reply:
I second @JackPiri's suggestion, but I cannot see it implemented in the code. Could you please double check?